diracx-db/src/diracx/db/sql/pilot_agents/db.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from datetime import datetime, timezone
+
+from sqlalchemy import insert
+
+from ..utils import BaseSQLDB
+from .schema import PilotAgents, PilotAgentsDBBase
+
+
+class PilotAgentsDB(BaseSQLDB):
+    metadata = PilotAgentsDBBase.metadata
+
+    async def addPilotReferences(
+        self,
+        pilotRef: list[str],
+        VO: str,
+        gridType: str = "DIRAC",
+        pilotStampDict: dict | None = None,
+    ) -> list[int]:
+
+        if pilotStampDict is None:
+            pilotStampDict = {}
+        row_ids = []
+        for ref in pilotRef:
+            stamp = ""
+            if ref in pilotStampDict:
+                stamp = pilotStampDict[ref]
+            now = datetime.now(tz=timezone.utc)
+            stmt = insert(PilotAgents).values(
+                PilotJobReference=ref,
+                VO=VO,
+                GridType=gridType,
+                SubmissionTime=now,
+                LastUpdateTime=now,
+                Status="Submitted",
+                PilotStamp=stamp,
+            )
+            result = await self.conn.execute(stmt)
+            row_ids.append(result.lastrowid)
```

Comment:
`lastrowid` is apparently not the preferred way to get the primary key (https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.CursorResult.lastrowid):
- `inserted_primary_key`: https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.CursorResult.inserted_primary_key
- or `returning()`: https://docs.sqlalchemy.org/en/20/core/dml.html#sqlalchemy.sql.expression.Insert.returning

Reply:
We don't need the return value, and if we choose a bulk insert it would not be possible anyway.